csv/client.go

```diff
@@ -0,0 +1,38 @@
+package csv
+
+type Options func(*Client)
+
+// Client is a csv client.
+type Client struct {
+	// fileFormat     string
+	IncludeHeaders bool
+	Delimiter      rune
+}
+
+func NewClient(options ...Options) (*Client, error) {
+	c := &Client{}
```

Comment:
Nit: `defaults()` is not needed here, you can just initialise the struct with the default values.

Reply:
Ok. Will remove. Why do we use that pattern in the plugins for initializing a client, but not here?

Reply:
because in plugins the spec is set via `unmarshal` so we don't have a place where we initialise the struct and we have to call a `setDefault` logic.